ballista/rust/core/src/serde/physical_plan/from_proto.rs

```diff
@@ -121,53 +120,64 @@ impl TryInto<Arc<dyn ExecutionPlan>> for &protobuf::PhysicalPlanNode {
             }
             PhysicalPlanType::CsvScan(scan) => {
                 let schema = Arc::new(convert_required!(scan.schema)?);
-                let options = CsvReadOptions::new()
-                    .has_header(scan.has_header)
-                    .file_extension(&scan.file_extension)
-                    .delimiter(scan.delimiter.as_bytes()[0])
-                    .schema(&schema);
                 let projection = scan.projection.iter().map(|i| *i as usize).collect();
-                Ok(Arc::new(CsvExec::try_new(
-                    &scan.path,
-                    options,
+                let statistics = convert_required!(scan.statistics)?;
+
+                Ok(Arc::new(CsvExec::new(
+                    Arc::new(LocalFileSystem {}),
+                    scan.files
+                        .iter()
+                        .map(|f| f.try_into())
+                        .collect::<Result<Vec<PartitionedFile>, _>>()?,
+                    statistics,
+                    schema,
+                    scan.has_header,
+                    str_to_byte(&scan.delimiter)?,
                     Some(projection),
                     scan.batch_size as usize,
-                    None,
-                )?))
+                    scan.limit.as_ref().map(|sl| sl.limit as usize),
+                )))
             }
             PhysicalPlanType::ParquetScan(scan) => {
-                let partitions = scan
-                    .partitions
-                    .iter()
-                    .map(|p| p.try_into())
-                    .collect::<Result<Vec<ParquetPartition>, _>>()?;
                 let schema = Arc::new(convert_required!(scan.schema)?);
                 let projection = scan.projection.iter().map(|i| *i as usize).collect();
+                let statistics = convert_required!(scan.statistics)?;
+
                 Ok(Arc::new(ParquetExec::new(
-                    partitions,
+                    Arc::new(LocalFileSystem {}),
```

Comment:
Same question here, are we planning to incorporate object_store both in DataFusion and Ballista? or just in Datafusion core for scope limit? I hit on your design doc on how to incorporate object_store into Ballista several days ago, it that the scope of another PR due to a separate registry passing mechanism of object stores?

Reply:
For now, the simplest solution I have to handle `ObjectStore` in Ballista is to do exactly the same way as for `TableProvider`:  define the implementations we support in a predefined enum of the protobuf and serialize/deserialize to/from that. I haven't created that enum for now as we have only 1 variant.

If we want to move away from that, we will need:
- to reference the ExecutionContext in the serde (requires some piping on the Ballista side)
- reorganize the ExecutionContext  into tiers as discussed in https://github.com/apache/arrow-datafusion/pull/1072#issuecomment-939727247 to avoid confusion
- find a way to dynamically load the same static configs accross all components of Ballista (with the challenge of adding new object stores / table providers **at build time**)